File: generated-code/example/csharp/Model/Enumeration.cs
namespace TransGenTest.Model
{
    public enum Enumeration
    {
        ValueOne = 0,
        ValueTwo = 1,
    }

    public static class EnumerationHelper {
        public static Enumeration ReadFrom(System.IO.BinaryReader reader) {
            switch (reader.ReadInt32())
            {
                case 0:
                    return Enumeration.ValueOne;
                case 1:
                    return Enumeration.ValueTwo;
                default:
                    throw new System.Exception("Unexpected tag value");
            }
        }
    }
}

File: generated-code/example/php/model/Enumeration.php
<?php

/**
 * Example enumeration
 */
abstract class Enumeration
{
    /**
     * First option
     */
    const VALUE_ONE = 0;

    /**
     * Second option
     */
    const VALUE_TWO = 1;

    /**
     * Read Enumeration from input stream
     */
    public static function readFrom($stream)
    {
        $result = $stream->readInt32();
        if (0 <= $result && $result < 2) {
            return $result;
        }
        throw new Exception('Unexpected tag value');
    }
}